integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ImageCatalogTests.java

```diff
@@ -0,0 +1,209 @@
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.isCreatedDeleted()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request then delete"
+        );
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(INVALID_IMAGECATALOG_URL), "same imagecatalog request again"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-delete-create");
+                }),  "check imagecatalog is created when name was used and deleted before");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testDeleteCbDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(DEFAULT_IMAGECATALOG_NAME)
+        );
+        when(ImageCatalog.delete(), "try to delete cb default image catalog");
+    }
+
+    @Test
+    public void testSetNewDeleteDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "default")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request and set as default"
+
+        );
+        when(ImageCatalog.post());
+        when(ImageCatalog.setDefault());
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "default");
+                    Assert.assertEquals(imageCatalog.getResponse().isUsedAsDefault(), true);
+                    try {
+                        given(ImageCatalog.request().withName(VALID_IMAGECATALOG_NAME + "default"));
+                        when(ImageCatalog.delete());
+                    } catch (Exception e) {
+                        LOGGER.error("Error occured during delete: " + e.getMessage());
+                    }
+                }),  "check imagecatalog is created, set as default and deleted");
+    }
+
+    @AfterSuite
+    public void cleanUp() throws Exception {
+        given(CloudbreakClient.isCreated());
+        when(ImageCatalog.getAll());
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Set<ImageCatalogResponse> imageCatalogResponses = imageCatalog.getResponses();
+                    for (ImageCatalogResponse response : imageCatalogResponses) {
+
+                        if (response.getName().startsWith("valid")) {
+                            try {
+                                given(ImageCatalog.request().withName(response.getName()));
```

Comment:
Please try not to use given/when statements in the then statement. Rather use the delete method